File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task45_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <iostream>
#include <limits>

int main() {
    int num, num2;
    std::cout << "Enter a number: ";
    while (!(std::cin >> num)) {
        std::cout << "Invalid input. Please enter a valid number.
";
        std::cout << "Enter a number: ";
        std::cin.clear();
        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '
');
    }
    std::cout << "You entered: " << num << std::endl;
    std::cout << "Enter another number: ";
    while (!(std::cin >> num2)) {
        std::cout << "Invalid input. Please enter a valid number.
";
        std::cout << "Enter another number: ";
        std::cin.clear();
        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '
');
    }
    std::cout << "You entered: " << num2 << std::endl;
    if (num2 == 0) {
        std::cout << "Cannot divide by zero.
";
        return 1;
    }
    int result = num / num2;
    std::cout << "Result: " << result << std::endl;
    return 0;
}

File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task45_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.util.InputMismatchException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        try {
            System.out.print("Enter a number: ");
            int num = scanner.nextInt();
            System.out.println("You entered: " + num);
            System.out.print("Enter another number: ");
            int num2 = scanner.nextInt();
            System.out.println("You entered: " + num2);
            int result = num / num2;
            System.out.println("Result: " + result);
        } catch (InputMismatchException e) {
            System.out.println("Invalid input. Please enter a valid number.");
        } catch (ArithmeticException e) {
            System.out.println("Cannot divide by zero.");
        } finally {
            scanner.close();
        }
    }
}